### PR TITLE
Fix licenses

### PR DESCRIPTION
**Pull request by mike1729**
# Description

This PR changes licensing: 
- all our products, i.e. crates under `bin` directory are licensed under GPL
- there rest is under Apache

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,7 +31,10 @@ If you would like to contribute, please fork the repository, introduce your chan
 
 ### License
 
-The code in this repository is licensed under the terms of the the Apache License 2.0.
+The code in this repository is licensed as follows:
+
+- all crates under `bin` directory are licensed under the terms of the GNU GPL version 3
+- all the rest crates are licensed under the terms of Apache License 2.0.
```
Comment (Marcin-Radecki) [inline README.md:37]: Nit: `all the rest crates` -> `rest of the crates`

File: aleph-client/Cargo.toml [hunk 2]
```diff
@@ -2,6 +2,8 @@
 name = "aleph_client"
 version = "1.4.0"
 edition = "2021"
+license = "Apache 2.0"
+license-file = "./LICENSE"
```
Comment (Marcin-Radecki) [inline aleph-client/Cargo.toml:6]: General comment to the whole PR: do we need to specify each license separately as a file? https://doc.rust-lang.org/cargo/reference/manifest.html#the-license-and-license-file-fields specifies we don't need both `license` and `license-file` to be set unless we're dealing with a known license from the list https://spdx.org/licenses/ , which is true in our case.

File: bin/cliain/Cargo.toml [hunk 3]
```diff
@@ -2,6 +2,8 @@
 name = "cliain"
 version = "0.4.0"
 edition = "2021"
+license = "GNU GPL 3.0"
```
Comment (Marcin-Radecki) [inline bin/cliain/Cargo.toml:5]: This should be this license https://spdx.org/licenses/GPL-3.0-only.html , and the corresponding string is ` GPL-3.0-only`

File: contracts/README.md [hunk 4]
```diff
@@ -1,71 +1,201 @@
...
-Specifically it will:
-- deploy the ERC20 token contract
-- deploy the game contract
-- transfer token balance to the game contract
-- whitelist some accounts for playing the game
-- interact with the game from the whitelisted account
-- wait past the game deadline, trigger game end and reward distribution
-- run an assert (simple check whether Pressiah received his reward according to the scheme). No more assertions are implemented at this point, as the hope is to soon start writing Rust-based tools for deployment and interacting with the contracts. 
-
+                                 Apache License
```
Comment (Marcin-Radecki) [inline contracts/README.md:1]: Looks like `README.md` got consumed :)

File: contracts/button_token/LICENSE [hunk 5]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (Marcin-Radecki) [inline contracts/button_token/LICENSE:1]: This crate seems to have `Cargo.toml`, we can set `license` attribute as well

File: contracts/yellow_button/LICENSE [hunk 6]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (Marcin-Radecki) [inline contracts/yellow_button/LICENSE:1]: This crate seems to have `Cargo.toml`, we can set `license` attribute as well

File: contracts/access_control/LICENSE [hunk 7]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (Marcin-Radecki) [inline contracts/access_control/LICENSE:1]: This crate seems to have `Cargo.toml`, we can set `license` attribute as well

**Review by Marcin-Radecki — CHANGES_REQUESTED** [review]

File: contracts/access_control/LICENSE (hunk 7 above)
Comment (mike1729) [reply contracts/access_control/LICENSE:1]: done

File: contracts/yellow_button/LICENSE (hunk 6 above)
Comment (mike1729) [reply contracts/yellow_button/LICENSE:1]: done

File: contracts/button_token/LICENSE (hunk 5 above)
Comment (mike1729) [reply contracts/button_token/LICENSE:1]: done

File: contracts/README.md (hunk 4 above)
Comment (mike1729) [reply contracts/README.md]: done

File: bin/cliain/Cargo.toml (hunk 3 above)
Comment (mike1729) [reply bin/cliain/Cargo.toml]: done

File: README.md (hunk 1 above)
Comment (mike1729) [reply README.md]: done

**Review by Marcin-Radecki — APPROVED** [review]

**Review by Yithis — APPROVED** [review]

**Pull request merged**